docs/language/ql-handbook/name-resolution.rst

```diff
@@ -61,31 +61,23 @@ following import statement::
 
     import examples.security.MyLibrary
 
-To find the precise location of this library module, the QL compiler processes the import 
+To find the precise location of this :ref:`library module <library-modules>`, the QL compiler processes the import 
 statement as follows:
 
   #. The ``.``\ s in the qualified reference correspond to file path separators, so it first looks 
      up ``examples/security/MyLibrary.qll`` from the directory containing ``Example.ql``. 
 
-  #. If that fails, it looks up ``examples/security/MyLibrary.qll`` relative to the enclosing query 
-     directory, if any. 
-     This query directory is a directory containing a |queries.xml file|_, and where the contents 
-     of that file is compatible with the current database schema.
-     (For example, if you are querying a JavaScript database, then the |queries.xml file|_ should 
-     contain ``<queries language="javascript"/>``.) 
+  #. If that fails, it looks up ``examples/security/MyLibrary.qll`` relative to the query 
+     directory, if any.
+     The query directory is the first enclosing directory containing a file called ``qlpack.yml``. (Or, in legacy products, a file called ``queries.xml``.)
```

Comment:
"First" is a bit ambiguous here -- but I'm not sure how to say it better without sounding excessively technical.